library/src/main/java/com/datatorrent/lib/io/fs/AbstractThroughputFileInputOperator.java

```diff
@@ -122,8 +123,18 @@ public void setPartitionCount(int requiredPartitions)
   @Override
   public void emitTuples()
   {
-    scanDirectory();
-
+    int fileCt = 0;
+    while ((!fileQueue.isEmpty()) && (fileCt < MAX_FILE_PER_WINDOW)) {
+      Path newPath = fileQueue.peek();
+      String newPathString = newPath.toString();
+      pendingFiles.add(newPathString);
```

Comment:
pendingFiles is serializable and used for recovery and idempotency, while fileQueue is not serialized and need to be recomputed upon failure and replay.  Moreover currently pending files is protected. It may cause regression issue if we change the whole design.